sashimi-webapp/src/logic/filemanager/data/folder.js

```diff
@@ -0,0 +1,128 @@
+import storage from 'src/database/storage';
+import File from './file';
+import idMap from './idmap';
+
+/* Error Messages */
+const ERROR_RENAME_FOLDER = 'Another folder in the current folder has the same name';
+const ERROR_RENAME_ROOTFOLDER = 'Root folder cannot be renamed';
+const ERROR_NOT_FOLDER_INSTANCE = '"this" is not an instance of "Folder"';
+
+/* Constants */
+const ORGANIZATION_ID = 1;
+const ROOT_FOLDER_ID = 0;
+
+/**
+* Folder Object
+*
+* @param {Number} folderID
+* @param {String} folderName
+* @param {String} folderPath
+*/
+export default function Folder(folderID, folderName, folderPath) {
+  this.id = folderID;
+  this.name = folderName;
+  this.path = folderPath;
+  this.parentFolder = null;
+  this.childFolderList = []; // list of folders in current folder
+  this.childFileList = []; // list of files in current folder
+}
+
+/* Private Functions */
+function hasSameFolderName(newFolderName) {
+  if (!(this instanceof Folder)) {
+    throw new Error(ERROR_NOT_FOLDER_INSTANCE);
+  }
+  const currParentFolder = this.parentFolder;
+  let currFolder;
+  let sameFolderName = false;
+  for (let i = 0; currParentFolder.childFolderList.length; i += 1) {
```

Comment:
just some code aesthetic idea, instead of `i` maybe can call it a more meaningful name like `folderIndex`